static/js/search.js

```diff
@@ -941,17 +942,27 @@ function parseInterval(interval) {
 }
 
 function timeChart() {
-    if (isTimechart) {
```

Comment:
If I run a query that supports a bar graph and then run another query that does not, the chart should go away and show the prompt `This query doesn't support visualization.`

Don't remove this code, we already have a condition check for the time chart query, so you can add the logic to check if it is an aggregation query.